flags/test.ts

```diff
@@ -736,3 +736,94 @@ Deno.test("valueFollowingDoubleHyphenIsNotUnknown", function (): void {
 Deno.test("whitespaceShouldBeWhitespace", function (): void {
   assertEquals(parse(["-x", "\t"]).x, "\t");
 });
+
+Deno.test("collectArgsDefaultBehaviour", function (): void {
+  const argv = parse([
+    "--foo",
+    "bar",
+    "--foo",
+    "baz",
+    "--beep",
+    "boop",
+    "--bool",
+    "--bool",
+  ]);
+
+  assertEquals(argv, {
+    foo: ["bar", "baz"],
```

Comment:
@c4spar 
Ah, I realized we still collect values for arg when there's no string, boolean, collect options are specified. What do you think about making this case `"baz"`?

Reply:
Because we check `hasCollect || hasTypes` before checking `collectable` in `setKey`, setting any option name changes the parse result (see the example below):
```
> parse(["--foo", "bar", "--foo", "baz"])
{ _: [], foo: [ "bar", "baz" ] }
> parse(["--foo", "bar", "--foo", "baz"], { string: "x" })
{ _: [], foo: "baz" }
```
I think this behavior is a little hard to explain to the users. What do you think?

Reply:
How about changing `hasCollect || hasTypes ? !collectable : (...)` part in `setKey` to just `!collectable`

Reply:
Also fine for me. I just thought that we want to keep the old behavior when no options are set, so that the flags module can be still used without defining any options. But i can change it if you want.
Current behaviour is roughly described here: https://github.com/denoland/deno_std/blame/0d2ba0f223d993fd31a0d10193de769ffb4a8886/flags/README.md#L69-L83 (But there is still an open question about negatable options.)

Reply:
> keep the old behavior when no options are set

Changing the parse behavior based on such condition feels too complex to me.

> so that the flags module can be still used without defining any options.

I think flags module is still usable without any options even if we removed the old behavior. Do you mean such change is too breaking to existing users?

Reply:
> I think flags module is still usable without any options even if we removed the old behavior. Do you mean such change is too breaking to existing users?

i mean also with collectible args. but i think i agree, it can be confusing. i will remove it.